### DISCUSSION TITLE
No Setpoint Device for Danfoss Thermostat DRS21

### DISCUSSION THREAD

**Discussion by ftsikogi** [zwave-js/zwave-js-ui#2834] [Q&A]
I am using zwave-js mqtt as a docker installation with Domoticz instead of Home Assistant. The problem that i am facing is irrelevant to Domoticz and it is relevant to how zwavejs intergrated with Danfoss Thermostat. It discovers 5 devices and it is missing the Setpoint Device. Please note that i am able to control the setpoint temperature through the zwavejs UI.
Please see the below links:

https://github.com/domoticz/domoticz/issues/5432

https://github.com/home-assistant/core/issues/46643

ZwaveJS: 10.3.0

Thanks for your help!

**Comment by robertsLando** [comment]
The problem like reported by kpine in [this](https://github.com/home-assistant/core/issues/46643#issuecomment-780002682) comment is that the device is not recognized as a climate. what you could do is to create a custom device config and add it to `store/customDevices.js` in a format like: https://github.com/zwave-js/zwave-js-ui/blob/master/hass/devices.ts

See [docs](https://zwave-js.github.io/zwave-js-ui/#/homeassistant/homeassistant-mqtt?id=custom-components)

**Reply by pquinqu** [reply]
Hi, so i tried to create an entire customDevices.js file for this DRS21 device, using the docs and the file format in devices.ts. When Zwave-js-ui tries to load the file it shows :
`zwave-js-ui  | 2023-02-05 11:41:42.867 ERROR GATEWAY: Failed to load /usr/src/app/store/customDevices.js: Unexpected token ':'
zwave-js-ui  | /usr/src/app/store/customDevices.js:6
zwave-js-ui  |   '2-32784-3': [
zwave-js-ui  |              ^
zwave-js-ui  |
zwave-js-ui  | SyntaxError: Unexpected token ':'
zwave-js-ui  |     at Object.compileFunction (node:vm:360:18)
...
zwave-js-ui  |     at loadCustomDevices (/usr/src/app/server/lib/Gateway.js:98:23)
zwave-js-ui  |     at Object.<anonymous> (/usr/src/app/server/lib/Gateway.js:124:1)
zwave-js-ui  | 2023-02-05 11:41:43.762 INFO APP: Version: 8.8.3.649eb88
zwave-js-ui  | 2023-02-05 11:41:43.767 INFO APP: Application path:/usr/src/app
`
Attached is the file.
[customDevices.zip](https://github.com/zwave-js/zwave-js-ui/files/10610782/customDevices.zip)

The doc says : `There, all components are stored that Z-Wave JS UI needs to create for each Z-Wave device type. The key is the Z-Wave device id(<manufacturerid>-<productid>-<producttype>). The value is an array with all Home Assistant components to be created for that Z-Wave device.`

For me the file seems to be valid. What did I do wrong ?
Thanks for your help. Phil

**Reply by robertsLando** [reply]
Giving that's a js file you should use `module.exports = ....`

```
module.exports = {
	'2-32784-3': [
		{
			type: 'climate',
			object_id: 'thermostat',
			values: [
....
```

**Reply by pquinqu** [reply]
Hi, thank you for the help: it works and i finally created the attached customDevice.js file to fix the case from ftsikogi.
[customDevices.zip](https://github.com/zwave-js/zwave-js-ui/files/10717400/customDevices.zip)
The file adds a climate entry for the Danfoss/Devolo DRS21 room thermostat. It works for my configuration (Z-Wave JS UI and Domoticz).

The doc says : "send us a PR!". I am afraid I don't know how to integrate that into the repo with a pull request. And in which file. I don't want to break anything.
So i guess an help would be great, if you have any time.
Thanks to you. Phil

**Reply by robertsLando** [reply]
@pquinqu I submitted a PR based on your attached file, thanks

https://github.com/zwave-js/zwave-js-ui/commit/de7ab0859e4a51b07b883c5c32bdf1edb717cca1
